### PR TITLE
ROX-13274: Scale down scanner default resource requests

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description
Scale down scanner default resource requests as too much CPU resources were requests on the data-plane cluster.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.

## Test manual

 - CI
 - Deploy Central with new default configuration

### CODE REVIEW

File: internal/dinosaur/pkg/defaults/default_scanner_spec.go [hunk 1]
```diff
@@ -2,18 +2,16 @@ package defaults
...
-	"github.com/caarlos0/env/v6"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 )
 
 // AnalyzerDefaults ...
 type AnalyzerDefaults struct {
 	MemoryRequest resource.Quantity `env:"MEMORY_REQUEST" envDefault:"100Mi"`
-	CPURequest    resource.Quantity `env:"CPU_REQUEST"    envDefault:"250m"`
+	CPURequest    resource.Quantity `env:"CPU_REQUEST"    envDefault:"10m"`
```
Comment (mtesseract) [inline internal/dinosaur/pkg/defaults/default_scanner_spec.go:12]: What we have applied was `5m` -- didn't you want to hard-code that config here?

File: internal/dinosaur/pkg/defaults/default_scanner_spec.go [hunk 2]
```diff
@@ -2,18 +2,16 @@ package defaults
...
 	MemoryLimit   resource.Quantity `env:"MEMORY_LIMIT"   envDefault:"2500Mi"`
```
Comment (mtesseract) [inline internal/dinosaur/pkg/defaults/default_scanner_spec.go:13]:
```suggestion
	MemoryLimit   resource.Quantity `env:"MEMORY_LIMIT"   envDefault:"2500M"`
```

File: internal/dinosaur/pkg/defaults/default_scanner_spec.go [hunk 3]
```diff
@@ -22,10 +20,10 @@ type AnalyzerDefaults struct {
 
 // DbDefaults ...
 type DbDefaults struct {
-	MemoryRequest resource.Quantity `env:"MEMORY_REQUEST" envDefault:"100Mi"`
-	CPURequest    resource.Quantity `env:"CPU_REQUEST"    envDefault:"250m"`
+	MemoryRequest resource.Quantity `env:"MEMORY_REQUEST" envDefault:"500Mi"`
```
Comment (mtesseract) [inline internal/dinosaur/pkg/defaults/default_scanner_spec.go:23]:
```suggestion
	MemoryRequest resource.Quantity `env:"MEMORY_REQUEST" envDefault:"500M"`
```

File: internal/dinosaur/pkg/defaults/default_scanner_spec.go [hunk 4]
```diff
@@ -22,10 +20,10 @@ type AnalyzerDefaults struct {
...
+	CPURequest    resource.Quantity `env:"CPU_REQUEST"    envDefault:"10m"`
 	MemoryLimit   resource.Quantity `env:"MEMORY_LIMIT"   envDefault:"2500Mi"`
```
Comment (mtesseract) [inline internal/dinosaur/pkg/defaults/default_scanner_spec.go:25]:
```suggestion
	MemoryLimit   resource.Quantity `env:"MEMORY_LIMIT"   envDefault:"2500M"`
```

**Review by mtesseract — APPROVED** [review]
I have corrected some defaults so that they are exactly what we have applied yesterday (see below).

Besides that, feel free to merge.

```json
{
  "central": {
    "resources": {
      "limits": {
        "cpu": "250m",
        "memory": "4G"
      },
      "requests": {
        "cpu": "50m",
        "memory": "250Mi"
      }
    }
  },
  "scanner": {
    "analyzer": {
      "scaling": {
        "autoScaling": "Enabled",
        "replicas": 1,
        "minReplicas": 1,
        "maxReplicas": 1
      },
      "resources": {
        "limits": {
          "cpu": "250m",
          "memory": "2500M"
        },
        "requests": {
          "cpu": "5m",
          "memory": "100M"
        }
      }
    },
    "db": {
      "resources": {
        "limits": {
          "cpu": "250m",
          "memory": "2500M"
        },
        "requests": {
          "cpu": "10m",
          "memory": "500M"
        }
      }
    }
  }
}
```

File: internal/dinosaur/pkg/defaults/default_scanner_spec.go (hunk 1 above)
Comment (SimonBaeumer) [reply internal/dinosaur/pkg/defaults/default_scanner_spec.go:12]: done

**Pull request merged**